master/internal/usergroup/api_groups.go

```diff
@@ -95,7 +116,18 @@ func (a *UserGroupAPIServer) GetGroup(ctx context.Context, req *apiv1.GetGroupRe
 		err = apiutils.MapAndFilterErrors(err)
 	}()
 
+	curUser, _, err := grpcutil.GetUser(ctx)
```

Comment:
If a user has zero group permissions ```GetGroup```, ```DeleteGroup```, ```UpdateGroup``` look like they will return access denied when performing them on a group that you do not have access to them.

We ideally want this to return the same exact response when performing with a group that does not exist so that the existence of a group isn't leaked to someone without permissions on that group.

In experiments this is handled by ```getExperiment``` returning ```canViewExperiment, serverError``` and returning the same 404 as the route would normally return when ```canViewExperiment``` is ```false```. All routes that use a specific experiment also do a call to ```getExperiment```